### ISSUE TITLE
NPM Warn deprecated reflect-metadata@0.2.1

### ISSUE THREAD

**Issue by CalebCyrill** [badeball/cypress-cucumber-preprocessor#1215]
### Current behavior
When installing the preprocessor I received the warning mesage ```npm WARN deprecated reflect-metadata@0.2.1: This version has a critical bug in fallback handling. Please upgrade to reflect-metadata@0.2.2 or newer.npm WARN deprecated reflect-metadata@0.2.1: This version has a critical bug in fallback handling. Please upgrade to reflect-metadata@0.2.2 or newer.``` A previously closed bug stated that this was handled in 20.0.5, but I am seeing it again in the latest version. 

### Desired behavior
The desired behavior is that it does not show the warning message that the deprecated reflect-metadata is still being used

### Versions

* **Cypress version**: 13.13.0
* **Preprocessor version**: 20.1.0
* **Node version**: 20.9.0

### Checklist

- [x] I've read the [FAQ](https://github.com/badeball/cypress-cucumber-preprocessor/blob/master/docs/faq.md).
- [x] I've read [instructions for logging issues](https://github.com/badeball/cypress-cucumber-preprocessor/blob/master/CONTRIBUTING.md#bug-reports).
- [x] I'm not using `cypress-cucumber-preprocessor@4.3.1` (package name has changed and it is no longer the most recent version, see [#689](https://github.com/badeball/cypress-cucumber-preprocessor/issues/689)).

**Comment by badeball** [comment]
The latest `@cucumber/gherkin` depends on an old version of `@cucumber/messages`, which in turn depends on `reflect-metadata@0.2.1`.

https://github.com/cucumber/gherkin/pull/252 fixes this, but has yet to be released.

**Comment by mikibakaiki** [comment]
I'm also getting the same warnings, with another one added: 

```
npm warn deprecated glob@7.2.3: Glob versions prior to v9 are no longer supported
npm warn deprecated glob@8.1.0: Glob versions prior to v9 are no longer supported
npm warn deprecated reflect-metadata@0.2.1: This version has a critical bug in fallback handling. Please upgrade to reflect-metadata@0.2.2 or newer.
npm warn deprecated reflect-metadata@0.2.1: This version has a critical bug in fallback handling. Please upgrade to reflect-metadata@0.2.2 or newer.
npm warn deprecated reflect-metadata@0.2.1: This version has a critical bug in fallback handling. Please upgrade to reflect-metadata@0.2.2 or newer.
```

**Comment by mikibakaiki** [comment]
seems that a [new version has been released ](https://github.com/cucumber/gherkin/releases/tag/v29.0.0) 😄

**Comment by badeball** [comment]
Said package isn't the only source of `reflect-metadata`, as seen below.

```
$ npm ls reflect-metadata
esbuild-ts@ /foobar/esbuild-ts
└─┬ @badeball/cypress-cucumber-preprocessor@20.1.2
  ├─┬ @cucumber/cucumber@11.0.0
  │ ├─┬ @cucumber/gherkin-utils@9.0.0
  │ │ └─┬ @cucumber/messages@24.1.0
  │ │   └── reflect-metadata@0.2.1
  │ └─┬ @cucumber/messages@24.1.0
  │   └── reflect-metadata@0.2.1
  ├─┬ @cucumber/gherkin@29.0.0
  │ └─┬ @cucumber/messages@25.0.1
  │   └── reflect-metadata@0.2.2 deduped
  └─┬ @cucumber/messages@26.0.0
    └── reflect-metadata@0.2.2
```

Tracking a bunch of transitive dependencies isn't something I'm going to spend my time on. If you have a request to update a specific dependency, then you can open up a ticket on this tracker. Before that, this is nothing but noise to me and not actionable.

**Comment by mikibakaiki** [comment]
Yeah, i agree it would be time consuming and not worth the effort! would you care to use [RenovateBot](https://docs.renovatebot.com/) as it would take care of most of all these things for you? :)

**Comment by badeball** [comment]
Dependencies will [automatically be updated](https://github.com/badeball/cypress-cucumber-preprocessor/issues/1290) once it can be, thus closing. If you find that a dependency has been updated but the change not yet released, you may open up an issue to request so.
